BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2335,12 +2362,21 @@ void AAMGuidance()
                 }
                 else// AAM Lead
                     aamTarget = MissileGuidance.GetAirToAirTarget(TargetPosition, TargetVelocity, TargetAcceleration, vessel, out timeToImpact, optimumAirspeed);
-
-
-                if (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f)
+                if(hasDataLink && (vrd != null && vrd.locked) && (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f))
                 {
-                    aamTarget = TargetPosition;
+                    if(Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight){
+                        TotalDrift += DataLinkDrift;
+                        aamTarget = MissileGuidance.GetDLDeviation(aamTarget,TotalDrift);
```

Comment:
Is this supposed to drift smoothly over time (i.e., actual random drift) or jump around from frame to frame (i.e., random noise)?
This implementation is doing the latter, but I suspect you want the former.
For the former, make `TotalDrift` a `Vector3` and use
```csharp
TotalDrift += DataLinkDrift * Time.fixedDeltaTime * VectorUtils.GaussianVector3();
aamTarget += TotalDrift;
```
then reset `TotalDrift` to `Vector3.zero` when needed.